android/src/main/java/io/ably/flutter/plugin/AblyEventStreamHandler.java

```diff
@@ -90,15 +96,12 @@ public void onCancel(Object object) {
             System.out.println("Cannot process null input on cancel");
             return;
         }
-        methodCallHandler.<AblyFlutterMessage<String>>ablyDo((AblyFlutterMessage)object, (ablyLibrary, message) -> {
-            String eventName = message.message;
-            switch (eventName) {
-                case PlatformConstants.PlatformMethod.onRealtimeConnectionStateChanged:
-                    ablyLibrary.getRealtime(message.handle).connection.off(connectionStateListener);
-                case PlatformConstants.PlatformMethod.onRealtimeChannelStateChanged:
-                    // ablyLibrary.getRealtime(handle).connection.off(connectionStateListener);
-            }
-        });
+        AblyFlutterMessage<String> message = getMessage(object);
+        String eventName = message.message;
+        switch (eventName) {
```

Comment:
With a single `case` perhaps this could be an `if` now. 😁 

Regardless, is it the intentional that this method is effectively a no-op if the `eventName` isn't exactly what is expected? Could we not consider failing here to indicate unexpected inbound?

Reply:
@QuintinWillison Yes, it can be an `if`, but soon we will have to switch back to `switch`.
And yes, it can be left as a no-op as this `cancel` event is triggered only when we are cancelling stream subscription. Necessary has been handled on the `onListen` method

Reply:
ok